### PR TITLE
Better camel->keyword 

### LINKED ISSUE #256: camel->keyword is incorrect for words with acronyms
Some examples of this problem:

```
;; #'amazonica.aws.identitymanagement/list-mfadevices
;; #'amazonica.aws.identitymanagement/list-open-idconnect-providers
;; #'amazonica.aws.identitymanagement/list-samlproviders
;; #'amazonica.aws.identitymanagement/list-sshpublic-keys
;; #'amazonica.aws.identitymanagement/list-virtual-mfadevices
;; #'amazonica.aws.iot/describe-cacertificate
;; #'amazonica.aws.iot/list-cacertificates
;; #'amazonica.aws.rds/describe-dbcluster-parameter-groups
;; #'amazonica.aws.rds/describe-dbcluster-snapshots
;; #'amazonica.aws.rds/describe-dbclusters
;; #'amazonica.aws.rds/describe-dbengine-versions
;; #'amazonica.aws.rds/describe-dbinstances
;; #'amazonica.aws.rds/describe-dbparameter-groups
;; #'amazonica.aws.rds/describe-dbsecurity-groups
;; #'amazonica.aws.rds/describe-dbsnapshots
;; #'amazonica.aws.rds/describe-dbsubnet-groups
```

I think it should instead be: `#"[A-Z]+(?=[A-Z]|$)|[A-Z][a-z]*"`.

### PR DESCRIPTION
**Pull request by lvh**
Closes #256. I have hopefully addressed all of @mcohen01's concerns around backwards compatibility via extensive tests. (I don't know how many people try to call almost every var in amazonica, but I'm one of them, so that's definitely a concern of mine!).

### CODE REVIEW

File: src/amazonica/core.clj [hunk 1]
```diff
@@ -284,13 +298,11 @@
...
          str/join)))
 
-
 (defn- aws-package?
   [clazz]
   (->> (.getName clazz)
        (re-find #"com\.amazonaws\.services")
-       nil?
-       not))
+       some?))
```
Comment (lvh) [inline src/amazonica/core.clj]: This is not germane to the PR, but I saw it, so I fixed it. (Per its own docstring, `some?` is considered more idiomatic.)

File: src/amazonica/core.clj [hunk 2]
```diff
@@ -920,10 +932,9 @@
                          (name fname) args)))))))
 
 (defn- client-methods
-  "Returns a map with keys of idiomatic Clojure hyphenated
-   keywords corresponding to the public Java method names of
-   the class argument, vals are vectors of
-   java.lang.reflect.Methods."
+  "Returns a map with keys of idiomatic Clojure hyphenated keywords
+  corresponding to the public Java method names of the class argument, vals are
+  vectors of java.lang.reflect.Methods."
```
Comment (lvh) [inline src/amazonica/core.clj]: Note: this didn't actually change except for the reflow. (It was different during an intermediate stage of the PR, but then I found a less intrusive way of making it work.)

File: test/amazonica/test/core.clj [hunk 3]
```diff
@@ -23,6 +25,8 @@
              elastictranscoder
              glacier
              identitymanagement
+             iot
+             iotdata
```
Comment (lvh) [inline test/amazonica/test/core.clj]: Not germane to the PR except in the sense that one of the nses I'm adding to is `iot`.

I could add code here to just require all of them programmatically, but I didn't want to cloud this PR.

**Comment by mcohen01** [comment]
This is awesome. Thanks Laurens. Super busy at the moment, but I will have time later today or tomorrow to look at it more closely and get this merged in.

**Comment by mcohen01** [comment]
Just leaving some other verification code here in case we ever want to compare again...

```
(def nms (-> (clojure.java.shell/sh "ls" "src/amazonica/aws/")
    :out
    (.split "\n")
    seq))

(defn print-vars [s]
  ;(println "-------------------------------")
  ;(println (symbol (str "amazonica.aws." (.replace s ".clj" ""))))
  ;(println "-------------------------------")
  (require [(symbol (str "amazonica.aws." (.replace s ".clj" "")))])
  (->> (str "amazonica.aws." (.replace s ".clj" ""))
       symbol
       find-ns
       ns-publics
       keys
       sort
       (map println)))

(map print-vars nms)


(clojure.set/difference
  (into #{}
        (-> (slurp "master.txt")
            (.split "\n")
            seq))
  (into #{}
        (-> (slurp "lvh.txt")
            (.split "\n")
            seq)))))
     
```

**Comment by mcohen01** [comment]
Thanks Laurens!!

**Comment by lvh** [comment]
awww yessss

**Comment by lvh** [comment]
I actually just pushed another test for ml-models (now there's no confusion between machine learning and multi-layer marketing -- do with that information what you will) but nbd :)

**Pull request merged**